core/test/base/TQObjectTests.cxx

```diff
@@ -0,0 +1,63 @@
+#include "gtest/gtest.h"
+#include "gmock/gmock.h"
+
+#include "Rtypes.h"
+#include "TQObject.h"
+#include "TQConnection.h"
+
+#include "RQ_OBJECT.h"
+
+// FIXME: We should think of a common place to put the mock objects.
+class TQConnectionMock : public TQConnection {
```

Comment:
Can you (to get us started) include a *very* simple test - something without mocks etc, but where you literally check whether say `TNamed::GetName()` returns what was set in the `TNamed` constructor?